app/models/alchemy/language.rb

```diff
@@ -99,6 +103,18 @@ def layout_root_page
 
     private
 
+    def set_locale
+      self.locale = ::I18n.available_locales.detect do |locale|
+        locale.to_s == code || locale.to_s == language_code
+      end
+    end
```

Comment:
Can you make this a public method `matching_locales` and display the select box ALSO if there's more than one matching locale?